addons/binding/org.openhab.binding.yamahareceiver2/src/main/java/org/openhab/binding/yamahareceiver2/handler/YamahaReceiverHandler.java

```diff
@@ -0,0 +1,197 @@
+        if (host_config != null && !host_config.equals(host)) {
+            host = host_config;
+            createCommunicationObject();
+        }
+
+        // Check if refresh configuration has changed
+        Integer interval_config = (Integer) thing.getConfiguration().get(YamahaReceiverBindingConstants.CONFIG_REFRESH);
+        if (interval_config != null && interval_config != refrehInterval) {
+            setupRefreshTimer(60 * 1000);
+        }
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void initialize() {
+        Configuration config = thing.getConfiguration();
+        host = (String) config.get(YamahaReceiverBindingConstants.CONFIG_HOST_NAME);
+
+        // Case: Lost host name somehow. This thing cannot be used any longer.
+        if (host == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Hostname not set!");
+            return;
+        }
+
+        createCommunicationObject();
+    }
+
+    /**
+     * We create a YamahaReceiverState that handles the current state (loudness, power, input etc)
+     * and a communication object.
+     */
+    private void createCommunicationObject() {
+        String zoneName = thing.getProperties().get(YamahaReceiverBindingConstants.CONFIG_ZONE);
+        if (zoneName == null) {
+            zoneName = YamahaReceiverCommunication.Zone.Main_Zone.name();
+        }
+
+        Zone zone = YamahaReceiverCommunication.Zone.valueOf(zoneName);
+
+        state = new YamahaReceiverState(new YamahaReceiverCommunication(host, zone));
+        try {
+            state.updateDeviceInformation();
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        }
+        updateReceiverState();
+        setupRefreshTimer(1000);
+
+        // If we are the main zone, detect other zones now.
+        if (zone == Zone.Main_Zone) {
+            zoneDiscoveryService = new ZoneDiscoveryService();
+            zoneDiscoveryService.start(bundleContext);
+            zoneDiscoveryService.detectZones(state, thing.getUID().getId());
+        }
+    }
+
+    private void setupRefreshTimer(int initial_wait_time) {
+        if (state == null) {
+            return;
+        }
+
+        Integer interval_config = (Integer) thing.getConfiguration().get(YamahaReceiverBindingConstants.CONFIG_REFRESH);
+
+        // Default is every 5min
+        if (interval_config == null) {
+            interval_config = new Integer(5);
+        }
+
+        if (refreshTimer != null) {
+            refreshTimer.cancel();
+        }
+        refreshTimer = new Timer();
+        refreshTimer.scheduleAtFixedRate(new TimerTask() {
+            @Override
+            public void run() {
+                updateReceiverState();
+            }
+        }, initial_wait_time, 60 * 1000 * interval_config);
```

Comment:
I will probably never understand why yamaha have choosed a full http rest interface with xml payload for their API on an embedded system. If requests are too frequent, the receiver drops some.
I wanted to avoid the situation that someone sets up a refresh of 1 sec, and some remote commands (from another app for example) won't go through because OH occupies the receiver.
But if the convention is, to use seconds for refresh if possible than I'll change that of course.